content/SCALE/SCALETutorials/Shares/NFS/AddingNFSShares.md

```diff
@@ -53,6 +53,12 @@ Leave the field empty to allow all systems access to the share.
 
 ### Adjusting Access Permissions
 
+{{< hint warning >}}
+You must set either `Maproot User` and `Maproot Group` OR `Mapall User` and `Mapall Group` to be able to write to an NFS share.
```

Comment:
This is incorrect.

Reply:
Could you please expand on this? Why is it incorrect?

When using a default install and creating an NFS share, after mounting, it failed to write any content until Mapall User and Mapall Group were set. This was using all default configurations and a fresh TrueNAS Scale installation.

Even if I'm off about the reason, it is still definitely the case that something either isn't documented or the defaults are broken / incomplete.

Reply:
Unless I'm misunderstanding you entirely, what you are describing is the way that sys auth works with NFS protocol. This is the reason why people use NIS, LDAP, AD, etc. C.f. man exports.

It doesn't work for you because the IDs on your client don't match with what's on the server.

Reply:
I'm fairly new to using NFS, and am not very familiar with how its authentication works. Is there documentation you would recommend to get a better understanding? I've heard of LDAP and AD, but not NIS or `C.f. man exports`. What are those? Google didn't dig up anything that looked pertinent.

Also, can the getting started documentation for TrueNAS be updated to guide users to either use the same IDs on the client and server, or use Mapall? I doubt I'm the only one who has tried to bootstrap NFS on a TrueNAS system, and gotten stumped when they get a non-descriptive permission denied error from nfs.